.github/workflows/yaml-checks.yml

```diff
@@ -0,0 +1,15 @@
+---
+name: yamllint
+on: [push, pull_request]
+
+jobs:
+  yamllint:
+    runs-on: ubuntu-latest
+    container:
+      image: registry.opensuse.org/devel/openqa/containers/os-autoinst_dev
+    steps:
+      - uses: actions/checkout@v4
+      - name: Validate yamls
+        run: |
+          git config --global --add safe.directory '*'
+          yamllint --strict $(git ls-files "*.yml" "*.yaml" 2> /dev/null || find . -name '*.y*ml')
```

Comment:
`*.y*ml` will match a lot of different things, maybe you want `find . -name '*.yaml' -o -name '*.yml'` to have closer match to the list used in git? Also `2> /dev/null` might be good addition for the find as well.